synapse/storage/databases/main/presence.py

```diff
@@ -12,13 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict, List, Tuple
+from typing import Dict, Iterable, List, Tuple
 
 from synapse.api.presence import UserPresenceState
 from synapse.storage._base import SQLBaseStore, make_in_list_sql_clause
 from synapse.util.caches.descriptors import cached, cachedList
+from synapse.util.constants import HOUR_IN_MS
 from synapse.util.iterutils import batch_iter
 
+# How long to keep entries in the users_to_send_full_presence_to db table
+USERS_TO_SEND_FULL_PRESENCE_TO_ENTRY_LIFETIME_MS = 14 * 24 * HOUR_IN_MS
```

Comment:
TBH I think it's fine to *not* have a timeout. At worst this table will have the same number of rows as the number in `users`.

Reply:
That's true - the rows are tiny and this won't include remote users. Oh, and I should probably make the `user_id` column a foreign key at the same time.

Reply:
Also if we don't have a timeout, it's likely not worth keeping `last_added_at_ms`.